### DISCUSSION TITLE
Репозиторий fct-search переименован в svodd

### DISCUSSION THREAD

**Discussion by audetv** [audetv/cohesion#3] [Announcements]
Произведен перенос репозитория fct-search на новый аккаунт организации terratensor.
Репозиторий fct-search переименован в svodd.
Технические работы по переносу успешно завершены.
Спасибо.

**Comment by iprst** [comment]
Благодарность!
Поздравляю коллектив с организацией.

**Comment by iprst** [comment]
После переноса сколько времени должно пройти для обновления системы и подхвата страниц в парсер? [Комментарий #441088](https://xn----8sba0bbi0cdm.xn--p1ai/qa/question/view-4295#:~:text=15:23%2010.06.2023) что-то уже час с лишним не появляется в большой своддной теме и поисковик его не видит.

**Reply by audetv** [reply]
Сразу должно было заработать, что-то не запустилось, сейчас буду проверять что.

**Reply by audetv** [reply]
Сейчас работает fct-parser так же перенёс, чтобы все сборки были из одного репозитория  (ghcr.io/terratensor), сервер не мог скачать сборку парсера и парсер просто не работал. Я его собирался перенести, но позже, но из за этого и не заработало, так как конфигурацию я сразу исправлял с расчетом, что все будет в одном репозитории.

**Reply by iprst** [reply]
Роджер (с)
Всё работает!

**Comment by audetv** [comment]
Как думаете, есть ли смысл перенести остальные issue связанные с концепцией из svodd?
Я перенес пару тем в cohesion #4 , #5 
или обратно их вернуть? не знаю даже. надо подумать как лучше сделать.

**Reply by iprst** [reply]
То что вы сделали — это прочитали мысли, я сам думал что нужно разобрать наши авгиевы конюшни перекрёстных issue (вроде даже одно обсуждение получилось перенести), но поскольку у вас был сегодня день большого переезда, решил пока отложить это, чтобы не напартачить чего-нибудь во время трансфера, этот процесс не виден, система не делает уведомлений, становится понятно по завершению переезда что модуль переехал. 

Да, нужно всё прошерстить и разобрать. Первое что хотел перенести, это [#39](https://github.com/terratensor/geomatrix/issues/39) — судя по всему парсер параграфов из html придётся вывести из геоматрицы, а issue открыть вновь, у меня набралось ещё три подобных источника, но все три с новым, другим оформлением, однако тоже однотипным и простым. Предлагаю сразу назвать это _small-parser_ или _light-parser_, а папку для входящих данных, которые он будет обрабатывать, назвать input или parse. 

ПС. подумал, скорее всего логичным названием будет _page-parser_ поскольку он парсит именно странички.

**Reply by audetv** [reply]
Или вообще эти темы перенести в дискуссии, так как они по сути пока не issue в техническом плане, которые могут быть выражены в коде и закрыты, а больше идеи и обсуждения. Но с другой стороны дискуссии у нас не особо прижились, хотя и в дискуссиях общались. не знаю. думаю, зачем-то об этом,

**Reply by iprst** [reply]
Тоже верно, думал об этом. У нас issue в половине случаев используются не по назначению, с другой стороны мы по сути используем недокументированные возможности, запрета на обсуждение внутри issue нет. Чем мне нравится issue, то что простая схема форматирования, в дискуссиях идёт «вопрос-ответ», это не всегда наглядно, в том числе потому что иногда черезчур наглядно.

**Reply by audetv** [reply]
> Да, нужно всё прошерстить и разобрать

Ага, хорошо, тогда будем разбирать) Вроде с переездом все. Еще есть сокращатель ссылок, но его позже перенесу. Кстати, а он работает? пошел проверять

**Reply by iprst** [reply]
> Еще есть сокращатель ссылок, но его позже перенесу. Кстати, а он работает?

Он работает. Заодно кое-что всплыло, что часто забивало глаз, но не доходило до оформления в проблему:
★ https://svodd.ru/rnOeLU6R

Как видим, ищем «Александр III», русского императора, однако конструкция с римской цифрой, то бишь тремя латинскими _I_, обрабатывается из рук вон: ищем императора, а подсветка подсвечивает всех Александр и Александров, в том числе авторов комментариев ))

Похоже, у нас словарь будет не только концептуальный, но ещё и «исторический» и «географический». Надо подумать, как это в реальности сделать. В мантикоре явно не хватает каких-то таких жизненных элементов.

**Reply by audetv** [reply]
> Кстати, а он работает? пошел проверять

Старые ссылки работают, значит все ок. Вообще, он и не должен был отвалиться, но раз уж произошел инцидент с парсером, то проверить надо) Он пока с моего репозитория и образ и код, потом перенесу, как буду делать какое-нибудь обновление в нем. не хочу сегодня больше переносов)

**Reply by iprst** [reply]
Ага, новые короткие ссылки тоже создаются и отдают правильный запрос после рендера в браузере.

**Reply by audetv** [reply]
> Тоже верно, думал об этом. У нас issue в половине случаев используются не по назначению, с другой стороны мы по сути используем недокументированные возможности, запрета на обсуждение внутри issue нет. Чем мне нравится issue, то что простая схема форматирования, в дискуссиях идёт «вопрос-ответ», это не всегда наглядно, в том числе потому что иногда черезчур наглядно.

Ну, значит надо оставить в том же виде, как складывается в процессе, перенести из дискуссий в issue и наоборот мы можем, если прямо возникнет необходимость. А так, продолжать в том же направлении, по своему произволу создавать дискуссии и issue. 
Приведем в порядок, перенесем концептуальные темы в cohesion.

> ПС. подумал, скорее всего логичным названием будет page-parser поскольку он парсит именно странички.

Да, надо будет собрать все обсуждения по парсеру и посмотреть, что получается. Технически это просто парсер с модулями, которые подключаются по мере поступления задач. Я бы его так сделал. но не сразу. Т.е. основной код - парсер а внутри папки источники которые он может обрабатывать, и сделать бы так ,что бы эти источники можно было добавлять, как в матрешку, - парсер фреймворк для поиска. Наверное, он таким и станет но не сразу. Парсер и в нём модули: сайт кремля, фкт, большая библиотека, тексты Пушкина и .т.д.

**Reply by audetv** [reply]
Просто по парсеру пока продолжаем и называем, как сейчас необходимо, а позже появится ядро и будет ясно, что с ним сделать и как эти куски соединить. И надо ли это делать. Хотя я уже в каждом новом парсере - пробе пера копирую вставляю одни и те же куски кода, они уже могут быть ядром для фреймворка.

**Reply by iprst** [reply]
С парсером наверное придётся балансировать, поскольку понятно что есть задачи здоровой разработки окружений, но параллельно есть мелкие задачи в духе «быстро обработать несколько тысяч файлов с известным параметром»: исходные файлы нет смысла хранить, они занимают кучу места, а после вывода нужной информации в csv их можно хотя бы сжать Ultra LZMA2 и забыть — у меня все библиотеки под которые вы делали bse-parser занимают 2% оригинального места после архивирования. Два, Карл! (с) 

На текущий момент это где-то 600 МБ вместо изначальных 30 Гб
В csv побольше, сейчас 2,12 Гб, но зато csv можно сразу забрасывать в базу.

**Reply by audetv** [reply]
> Как видим, ищем «Александр III», русского императора, однако конструкция с римской цифрой, то бишь тремя латинскими _I_, обрабатывается из рук вон: ищем императора, а подсветка подсвечивает всех Александр и Александров, в том числе авторов комментариев ))

Надо из обычного режима поиска убрать поиск по имени автора,  т.е. по идее поиск должен искать только в тексте, в параграфах, а остальное надо подключать фильтрами по необходимости. Как это уже оформить, чтобы без проблем мог пользоваться пользователь другой вопрос. Я имею ввиду, что при запросе поисковом указывается в структуре запроса(технической json) только поле text(paragraph) и не производить поиск в других полях без явного на то указания. Ещё когда аватар делали, тогда уже эта ситуация с пользователями мне не нравилась. Например, даже если просто поставить галочку в настройках поиска: искать по пользователям, уже будет другая выдача, галочку подобную, как концептуальный поиск. Но ее нажимать надо) Я думаю мы к пользователям еще раз придем, когда начнем подключать другие источники. мы же хотим объединять это в один поиск независимо от источника. Вот когда с толстыми книгами начнем, как сделаю фронтенд отдельный, просто для понимания и потестировать, и дальше посмотрим, как соединит это и возможно начнёт выявлятся система и пользователей определим куда-нибудь.

**Reply by iprst** [reply]
> Надо из обычного режима поиска убрать поиск по имени автора, т.е. по идее поиск должен искать только в тексте, в параграфах, а остальное надо подключать фильтрами по необходимости. Как это уже оформить, чтобы без проблем мог пользоваться пользователь другой вопрос. Я имею ввиду, что при запросе поисковом указывается в структуре запроса(технической json) только поле text(paragraph) и не производить поиск в других полях без явного на то указания. Ещё когда аватар делали, тогда уже эта ситуация с пользователями мне не нравилась.

Да. Меня тоже посещали мысли на эту тему. Что сейчас круто — текстовый поиск реализуется одновременно с аватарой, то есть можно найти такой-то текст такого-автора, это очень полезно. Нужно сохранить это поведение, выведя поиск и подсветку из поля «автор», и по этому полю искать уже специальным образом, который сейчас задаётся сценарием «аватара в строке».

**Reply by audetv** [reply]
> С парсером наверное придётся балансировать

Да, по сути они различаются на те которые постоянно будут парсить контент онлайн: ФКТ, различные сайты и т.д. и те, которые используем пару раз в процессе подготовки материалов. Вторые как раз можно объединить в один репозиторий и пусть лежат по разным папкам. в каждой можно сделать readme со ссылкой на issue историю появления, зачем парсер, и пусть лежат в архиве.
По идее, ко вторым можно отнести и парсер для толстых книг, он по сути так же нужен один раз и потом будет лежать в архиве.

**Reply by iprst** [reply]
Да, логично. Парсер для обновляемых отслеживаемых сайтов веб2.0 это одно направление, парсер для типовых статичных материалов (книг, страниц) это другое направление.
